File: frontend/buddease/src/app/projects/DataAnalysisPhase/DataAnalysis.ts
// models/dataAnalysis/DataAnalysis.ts

export class DataAnalysis {
    id: number;
    name: string;
    result: string;
    description: string;
    file_path: string;
  
    constructor(id: number, name: string, result: string, description: string, file_path: string) {
      this.id = id;
      this.name = name;
      this.result = result;
      this.description = description;
      this.file_path = file_path;
    }
  }
  

File: backend/models/data/data_analysis.py
# models.py
from flask_sqlalchemy import SQLAlchemy

db = SQLAlchemy()

class DataAnalysis(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(255))
    result = db.Column(db.Text)  # Assuming 'result' is a field you want to include
    description = db.Column(db.Text)
    file_path = db.Column(db.String(255))

    def __init__(self, id, name, result, description, file_path):
        self.id = id
        self.name = name
        self.result = result
        self.description = description
        self.file_path = file_path

    def __repr__(self):
        return f"<DataAnalysis(id={self.id}, name={self.name}, result={self.result}, description={self.description}, file_path={self.file_path})>"
